crypto/s2n_libcrypto.c

```diff
@@ -226,3 +226,12 @@ bool s2n_libcrypto_supports_flag_no_check_time()
     return false;
 #endif
 }
+
+bool s2n_libcrypto_supports_providers(void)
+{
+#ifdef S2N_LIBCRYPTO_SUPPORTS_PROVIDERS
```

Comment:
Probably not that concerning, but if `S2N_LIBCRYPTO_SUPPORTS_PROVIDERS` is somehow defined to be 0, it probably wouldn't be great to have a mix of ifs and ifdefs.
```suggestion
#if S2N_LIBCRYPTO_SUPPORTS_PROVIDERS
```